examples/ErrorHandling/HandleKeywordPolicyViolations.php

```diff
@@ -0,0 +1,261 @@
+                $options[ArgumentNames::AD_GROUP_ID] ?: self::AD_GROUP_ID,
+                $options[ArgumentNames::KEYWORD_TEXT] ?: self::KEYWORD_TEXT
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID without hyphens
+     * @param int $adGroupId the ad group ID to add a keyword to
+     * @param string $keywordText the keyword text to add
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId,
+        string $keywordText
+    ) {
+        // Configures the keyword text and match type settings.
+        $keywordInfo = new KeywordInfo([
+            'text' => new StringValue(['value' => $keywordText]),
+            'match_type' => KeywordMatchType::EXACT
+        ]);
+
+        // Constructs an ad group criterion using the keyword text info above.
+        $adGroupCriterion = new AdGroupCriterion([
+            'ad_group' => new StringValue(
+                ['value' => ResourceNames::forAdGroup($customerId, $adGroupId)]
+            ),
+            'status' => AdGroupCriterionStatus::ENABLED,
+            'keyword' => $keywordInfo
+        ]);
+
+        $adGroupCriterionOperation = new AdGroupCriterionOperation();
+        $adGroupCriterionOperation->setCreate($adGroupCriterion);
+        $adGroupCriterionServiceClient = $googleAdsClient->getAdGroupCriterionServiceClient();
+
+        $exemptPolicyViolationKeys = [];
+        try {
+            // Try sending a mutate request to add the keyword.
+            $response = $adGroupCriterionServiceClient->mutateAdGroupCriteria(
+                $customerId,
+                [$adGroupCriterionOperation]
+            );
+            printf(
+                "Added a keyword with resource name '%s'.%s",
+                $response->getResults()[0]->getResourceName(),
+                PHP_EOL
+            );
+            return;
```

Comment:
It may be useful to add a comment clarifying that this return statement is never going to be executed because the keyword creation will throw an exception, it's not really easy to follow otherwise.

Reply:
I added a comment to explain more. Actually this line will be executed if you run the example more than once.